### DISCUSSION TITLE
Contextual dates

### DISCUSSION THREAD

**Discussion by jenthomson** [ministryofjustice/moj-frontend#292] [EXPERIMENTAL: Components, pages and patterns]
![image](https://user-images.githubusercontent.com/1866734/147359799-4eebf319-25d3-45e3-b080-7ae7909055aa.png)

## Existing

have not come across any existing similar patterns or Github discussions

## When to use
Use the contextual date pattern to indicate important information about a date in a table to make it easier to scan, interpret and action where required.

### Writing content
The date can be written as DD/MM/YY or DD Month YYYY, depending on the use case of the system

The label should be short and succinct. Recommend no more than 20 characters.

Never use this component without a label - the label provides the context to the visual highlight for accessibility.

### Making sure it’s accessible

- Recommended typography is standard black text
-- Bold 19px for the date
-- Regular 16px for the context 
- The highlight line should have appropriate colour contrast
- When using for to communicate more than one context the highlight line should use colours that are not too close to each other for most differentiation
- Never use this component without a label - the label provides the context to the visual highlight for accessibility.
 

### Implementing the code
- Use spans to wrap the elements –should be stacked on top of each other
- The numeric part should be 19px and the label 16px
- Text should be left aligned

Using it in context with other components and as part of a service?
Use flex-box to display multiple instances in one row

**Using it in context with other components and as part of a service?**
- Original use case is within a table. Not been tried in other context yet.
 

### When to use variations? (if applicable)

First iteration will have only one variation

## Things we don’t know enough about

- Linking to the detail or filtering data
- Different layout, size or colour combinations
-

**Comment by helennickols** [comment]
The contextual date has been published on the MOJ Design System. This is to make it easier to find and use.

https://design-patterns.service.justice.gov.uk/components/contextual-date/
